File: online-judge-solutions/Codeforces/1709A.py
#https://codeforces.com/problemset/problem/1709/A
#Lang		:	Python 3.8
#Time		:	31 ms
#Memory         :	0 KB
for _ in range(int(input())):
    x,m=int(input()),{}
    a,b,c=map(int,input().split())
    m[0],m[1],m[2],m[3]=0,a,b,c
    print("YES" if m[m[x]] else "NO")

File: online-judge-solutions/Codeforces/1709A.cpp
//https://codeforces.com/problemset/problem/1709/A
//Lang		:	GNU C++14
//Time		:	0 ms
//Memory	:	0 KB
#include<iostream>
#include<map>
using namespace std;
int main(){
	int t,a,b,c,x;
	cin>>t;
	while(t--){
		map<int,int> m;
		cin>>x>>a>>b>>c;
		m[1]=a;
		m[2]=b;
		m[3]=c;
		cout<<(m[m[x]]?"YES":"NO")<<"\n";
	}
	return 0;
}
